### PR TITLE
[SQL Migration] Improve IR registration experience

### PR DESCRIPTION
**Pull request by raymondtruong**
This PR:
-  adds IR registration instructions to the IR selection page (and DMS details dialog). Previously, the instructions were only shown in the dialog where the DMS service is created, meaning a user with an already created but unregistered DMS service would not be shown instructions, leading to confusion.
- implements refresh authentication key functionality in all three places where the authentication key table is shown (IR selection page, DMS creation dialog, DMS details dialog), and extracts the instructions/table generation to a shared helper for consistent behavior
- adds additional text to the connection status box of already-registered DMS services for benefits of registering additional nodes
- adds clarifying instructions for when IR registration instructions will be shown on service creation (only for network share/DB scenarios), as IR registration instructions do not appear when creating a new service for blob scenarios
- fixes minor typos and updates IR download link to an aka.ms link

Screenshots (see highlighted sections): 

First-time registration:

image

Registering additional nodes:

image

Creating new DMS service:

image

DMS details dialog:

image

### CODE REVIEW

**Review by aperfilev — APPROVED** [review]
LGTM

File: extensions/sql-migration/src/constants/strings.ts [hunk 1]
```diff
@@ -770,7 +774,7 @@ export function SERVICE_ERROR_NOT_READY(serviceName: string, error: string): str
 		error);
 }
 export function SERVICE_READY(serviceName: string, host: string): string {
-	return localize('sql.migration.service.ready', "Azure Database Migration Service '{0}' is connected to self-hosted integration runtime running on the node - {1}", serviceName, host);
+	return localize('sql.migration.service.ready', "Azure Database Migration Service '{0}' is connected to self-hosted integration runtime running on node(s) - {1}\n\nFor improved performance and high Availability, you can register additional nodes. See below for registration instructions.", serviceName, host);
 }
```
Comment (aperfilev) [inline extensions/sql-migration/src/constants/strings.ts:778]: lower case?

File: extensions/sql-migration/src/constants/strings.ts [hunk 2]
```diff
@@ -740,13 +740,17 @@ export const MIGRATION_SERVICE_NAME_INFO = localize('sql.migration.services.name
...
 export const LOADING_MIGRATION_SERVICES = localize('sql.migration.service.container.loading.help', "Loading Migration Services");
-export const SERVICE_CONTAINER_HEADING = localize('sql.migration.service.container.heading', "Setup integration runtime");
+export const SERVICE_CONTAINER_HEADING = localize('sql.migration.service.container.heading', "Set up integration runtime");
 export const SERVICE_CONTAINER_DESCRIPTION1 = localize('sql.migration.service.container.container.description1', "Azure Database Migration Service leverages Azure Data Factory's self-hosted integration runtime to upload backups from on-premises network file share to Azure.");
-export const SERVICE_CONTAINER_DESCRIPTION2 = localize('sql.migration.service.container.container.description2', "Follow the instructions below to setup self-hosted integration runtime.");
+export const SERVICE_CONTAINER_DESCRIPTION2 = localize('sql.migration.service.container.container.description2', "Follow the instructions below to set up self-hosted integration runtime.");
```
Comment (brian-harris) [inline extensions/sql-migration/src/constants/strings.ts:745]: should this be 'set up' or 'setup'?

File: extensions/sql-migration/src/constants/strings.ts (hunk 1 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/constants/strings.ts:778]: DMS is the official product name, will leave it capitalized

File: extensions/sql-migration/src/constants/strings.ts (hunk 2 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/constants/strings.ts:745]: "Set up" - "setup" is the noun/adjective, "set up" is more correct for the act of setting up :)

File: extensions/sql-migration/src/constants/strings.ts (hunk 1 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/constants/strings.ts:778]: ah, I'm assuming you meant "Availability" - will fix that, good catch

File: extensions/sql-migration/src/wizard/integrationRuntimePage.ts [hunk 3]
```diff
@@ -787,3 +783,68 @@ export function createAuthenticationKeyTable(view: azdata.ModelView,): azdata.De
...
+
+	const irSetupStep1Text = view.modelBuilder.text().withProps({
+		value: constants.SERVICE_STEP1,
+		CSSStyles: {
+			...styles.BODY_CSS
+		},
+		links: [
+			{
+				text: constants.SERVICE_STEP1_LINK,
+				url: 'https://www.microsoft.com/download/details.aspx?id=39717'
```
[the remark this answers is not in the document]
Comment (raymondtruong) [reply extensions/sql-migration/src/wizard/integrationRuntimePage.ts:817]: was thinking the same - we should slowly start replacing our hardcoded links with aka.ms links. Will create one now for this one (if it doesn't already exist)

**Review by AkshayMata — APPROVED** [review]

**Review by brian-harris — APPROVED** [review]
looks awesome !!!

**Pull request merged**